pass down -Dzlib.prefix mvn option as -DZLIB_ROOT to cmake used for building native Hadoop libraries

fixes "Could NOT find ZLIB (missing: ZLIB_INCLUDE_DIR)" error when zlib headers are not installed in OS

author: Kenneth Hoste (HPC-UGent)
--- hadoop-2.9.2-src/hadoop-common-project/hadoop-common/pom.xml.orig	2019-11-28 10:20:59.625554136 +0000
+++ hadoop-2.9.2-src/hadoop-common-project/hadoop-common/pom.xml	2019-11-28 10:21:20.875786582 +0000
@@ -626,6 +626,7 @@
                     <CUSTOM_SNAPPY_PREFIX>${snappy.prefix}</CUSTOM_SNAPPY_PREFIX>
                     <CUSTOM_SNAPPY_LIB>${snappy.lib} </CUSTOM_SNAPPY_LIB>
                     <CUSTOM_SNAPPY_INCLUDE>${snappy.include} </CUSTOM_SNAPPY_INCLUDE>
+                    <ZLIB_ROOT>${zlib.prefix}</ZLIB_ROOT>
                     <REQUIRE_ZSTD>${require.zstd}</REQUIRE_ZSTD>
                     <CUSTOM_ZSTD_PREFIX>${zstd.prefix}</CUSTOM_ZSTD_PREFIX>
                     <CUSTOM_ZSTD_LIB>${zstd.lib}</CUSTOM_ZSTD_LIB>
